contrib/database/sql/conn.go

```diff
@@ -203,5 +203,8 @@ func (tp *traceParams) tryTrace(ctx context.Context, qtype queryType, query stri
 			span.SetTag(k, v)
 		}
 	}
-	span.Finish(tracer.WithError(err))
+	defer span.Finish()
+	if err != nil && tp.cfg.errCheck(err) {
```

Comment:
I think I'd feel safer from future programming errors if we'd check for `nil` here:
```suggestion
	if err != nil && tp.cfg.errCheck != nil && tp.cfg.errCheck(err) {
```
It's a bit uglier, but it makes the code `OpenDB` easier.

Ultimately, if the default non-nil value would be set in `(contrib/database/sql/option.go).defaults` just like all the other integrations, we'd need none of it.

Reply:
Yes, at first, I set default function in `defaults`.
But in this case, we need to do like that check `cfg.errCheck` is default value or not, and if it is default value, we need to check `rc.errCheck` is default value or not for fallbacking at [here](https://github.com/DataDog/dd-trace-go/pull/1315/files#diff-2f8e1b87e4bb1c3e07f64731592b0e997ce8430fbdffd2109436ff176133ea01R189-R195).
But function value is hard to detect is it default value or not.

It is reason that I stopped to fill in defaults. So checking `tp.cfg.errCheck` is not nil before calling is good, I think.